api/types/id_response.go

```diff
@@ -9,5 +9,5 @@ type IDResponse struct {
 
 	// The id of the newly created object.
 	// Required: true
-	ID string `json:"Id"`
+	ID string `json:"ID"`
```

Comment:
I don't think we can / should change this one. While it's unfortunate that this was used as `Id` instead of `ID`, changing it now would likely be a breaking change.


It looks like the Secrets and Configs create endpoints "fixed" the wrong naming, but indeed are not consistent. Both use their own type for the response;

https://github.com/moby/moby/blob/de9f12ea962608bd257a07af72a792eba2a38132/api/types/types.go#L736-L741

https://github.com/moby/moby/blob/de9f12ea962608bd257a07af72a792eba2a38132/api/types/types.go#L748-L753

So perhaps we need a separate definition for them in the swagger as well. Alternatively, we could have a "legacyIDresponse" and an IDRsponse", but not sure if we should go that route.

Reply:
Thanks! It's definitely unfortunate we ended up with differences in these (some of that originates from _long time ago_, so we may be kinda "stuck" with them).

Reply:
(and it also doesn't help that Go's JSON marshalling / unmarshaling is mostly case-insensitive 😓)

Reply:
@thaJeztah PR is updated with the new swagger types for SecretCreateResponse/ConfigCreateResponse 😄